src/social-login/services/social-login.service.ts

```diff
@@ -54,75 +57,60 @@ export class SocialLoginService {
     }
     return { authUrl };
   }
-
-  async socialLogin(req) {
-    Logger.log('socialLogin() starts', 'SocialLoginService');
+  async socialLogin(req): Promise<{
+    mfaRequired: boolean;
+    refreshToken?: string;
+    accessToken?: string;
+    authenticators?: string[];
+    sessionId?: string;
+  }> {
+    Logger.log(
+      'Inside handleGoogleLogin() to create or fetch user detail based on login',
+      'SocialLoginService',
+    );
     const { email, name, profileIcon } = req.user;
-    const rawUrl = this.config.get('INVITATIONURL');
-    const url = new URL(rawUrl);
-    const domain = url.origin;
-    let userInfo = await this.userRepository.findOne({
+    let user = await this.userRepository.findOne({
       email,
     });
-    let appUserID;
-    if (!userInfo) {
-      appUserID = `${Date.now()}-${uuidv4()}`;
-      // Giving default access of services...
+    if (!user) {
+      const userId = `${Date.now()}-${uuidv4()}`;
       const ssiAccessList = this.supportedServiceList.getDefaultServicesAccess(
         SERVICE_TYPES.SSI_API,
       );
       const kycAccessList = this.supportedServiceList.getDefaultServicesAccess(
         SERVICE_TYPES.CAVACH_API,
       );
-      // const questAccessList =
-      //   this.supportedServiceList.getDefaultServicesAccess(SERVICE_TYPES.QUEST);
-      userInfo = await this.userRepository.create({
+      user = await this.userRepository.create({
         email,
-        userId: appUserID,
-        name: name,
+        userId,
+        name,
         profileIcon,
         accessList: [...ssiAccessList, ...kycAccessList],
-        role: UserRole.ADMIN,
       });
-    } else {
-      const updates: Partial<UserDocument> = {};
-      if (!userInfo.name) updates.name = name;
-      if (!userInfo.profileIcon) updates.profileIcon = profileIcon;
-      if (Object.keys(updates).length > 0) {
-        this.userRepository.findOneUpdate({ email }, updates);
-      }
     }
-    Logger.log('socialLogin() starts', 'SocialLoginService');
-
-    let isVerified = false;
-    let authenticator = null;
-    if (userInfo.authenticators && userInfo.authenticators.length > 0) {
-      authenticator = userInfo.authenticators?.find((x) => {
-        if (x && x.isTwoFactorAuthenticated) {
-          return x;
-        }
-      });
-      isVerified = authenticator
-        ? authenticator.isTwoFactorAuthenticated
-        : false;
+    const updates: Partial<UserDocument> = {};
+    if (!user.name) updates.name = name;
+    if (!user.profileIcon) updates.profileIcon = profileIcon;
+    if (Object.keys(updates).length > 0)
+      this.userRepository.findOneUpdate({ email }, updates);
+    const sessionId = await this.createSession(user);
+    const activeAuthenticators = user.authenticators?.filter(
+      (a) => a.isTwoFactorAuthenticated,
```

Comment:
```suggestion
      (a) => a.isTwoFactorEnabled,
```
According to seq diagram 
<img width="474" height="171" alt="image" src="https://github.com/user-attachments/assets/c7bdac80-3804-4f80-b75e-b75b645648f9" />
We are checking is user has mfa enabled or not.

Here  in the code we are not doing not any enabled check we are filtering out how many authenticaors we have enabled or setup.